csharp/src/Apache.Arrow/Ipc/ArrowStreamReader.cs

```diff
@@ -14,282 +14,61 @@
 // limitations under the License.
 
 using System;
-using System.Buffers;
-using System.Collections.Generic;
-using System.Diagnostics;
 using System.IO;
-using System.Linq;
 using System.Threading;
 using System.Threading.Tasks;
 
 namespace Apache.Arrow.Ipc
 {
+    /// <summary>
+    /// Represents a reader that can read Arrow streams.
+    /// </summary>
     public class ArrowStreamReader : IArrowReader, IDisposable
     {
-        public Schema Schema { get; protected set; }
-        public Stream BaseStream { get; }
+        private protected readonly ArrowReaderImplementation _implementation;
 
-        protected ArrayPool<byte> Buffers { get; }
+        public Schema Schema => _implementation.Schema;
 
         public ArrowStreamReader(Stream stream)
```

Comment:
I wonder if this should instead be a factory method, such that if, for example, it was given a FileStream, it could then return the implementation optimized for files, or if a MemoryStream, the one for memory, etc?